charmhelpers/contrib/openstack/context.py

```diff
@@ -478,6 +518,17 @@ def __call__(self):
                         'service_project_id': rdata.get('service_tenant_id'),
                         'service_domain_id': rdata.get('service_domain_id')})
 
+                # NOTE:
+                # keystone-k8s operator presents full URLS
+                # for all three endpoints - public and internal are
+                # externally addressable for machine based charm
+                if 'public_auth_url' in rdata:
+                    ctxt.update({
+                        'public_auth_url': rdata.get('public_auth_url'),
+                        'internal_auth_url': rdata.get('internal_auth_url'),
```

Comment:
Is it worth checking for `internal_auth_url` separately and adding it; e.g. this would KeyError if not available?